File: Data Structures & Algorithms/binary-tree-diameter/submission-7.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private int result = 0;
    private int getHeight(TreeNode root)
    {
        if(root == null)
        {
            return 0;
        }
        int left = getHeight(root.left);
        int right = getHeight(root.right);
        result = Math.max(result, left + right);
        return 1 + Math.max(left, right);
    }
    public int diameterOfBinaryTree(TreeNode root) {
        getHeight(root);
        return result;

    }
}

File: Data Structures & Algorithms/valid-binary-search-tree/submission-7.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        queue = deque([(root,-1001, 1001)])
        while queue:
            length = len(queue)
            for _ in range(length):
                nd, l, r = queue.popleft()
                if nd.val <= l or nd.val >= r:
                    return False
                if nd.left:
                    queue.append((nd.left,l, nd.val))
                if nd.right:
                    queue.append((nd.right, nd.val, r))
        return True